plugins/module_utils/foreman_helper.py

```diff
@@ -228,6 +230,12 @@ def __init__(self, argument_spec, **kwargs):
         self.state = 'undefined'
         self.entity_spec = {}
 
+    @contextmanager
+    def start(self):
```

Comment:
I think I would have called it `api_connection`, so you end up with (eh), `with module.api_connection(): …`

Reply:
You can make the module itself a context manager so you would have 'with module:'.

Reply:
Yes, but I wanted to avoid making any connection to the server before validating `entity_dict` params that can be validated before opening connection, like it is in some modules.